en/lessons/ecto/basics.md

```diff
@@ -0,0 +1,232 @@
+---
+version: 2.0.0
+title: Basics
+---
+
+Ecto is an official Elixir project providing a database wrapper and integrated query language.
+With Ecto we're able to create migrations, define schemas, insert and update records, and query for them.
+
+{% include toc.html %}
+
+### Adapters
+
+Ecto supports different databases through the use of adapters.  A few examples of adapters are:
+
+* PostgreSQL
+* MySQL
+* SQLite
+
+For this lesson we'll configure Ecto to use the PostgresQL adapter.
+
+### Getting Started
+
+Through the course of this lesson we'll cover three parts to Ecto:
+
+* The Repository — provides the interface to our database, including the connection
+* Migrations — a mechanism to create, modify, and destroy database tables and indexes
+* Schemas — specialized structs that represent database table entries
+
+To start we'll create an application with a supervision tree.
+
+```shell
+$ mix new example --sup
+$ cd example
+```
+
+Add the ecto and postgrex package dependencies to your `mix.exs` file.
+
+```elixir
+  defp deps do
+    [
+      {:ecto, "~> 2.0"},
+      {:postgrex, "~> 0.11"}
+    ]
+  end
+```
+
+Fetch the dependencies using
+
+```shell
+$ mix deps.get
+```
+
+#### Creating a Repository
+
+A repository in Ecto maps to a datastore such as our Postgres database.
+All communication to the database will be done using this repository.
+
+Set up a repository by running:
+
+```shell
+$ mix ecto.gen.repo -r Example.Repo
+```
+
+This will generate the configuration required in `config/config.exs` to connect to a database including the adapter to use.
+This is the configuration file for our `Example` application
+
+```elixir
+config :example, Example.Repo,
+  adapter: Ecto.Adapters.Postgres,
+  database: "example_repo",
+  username: "postgres",
+  password: "",
+  hostname: "localhost"
+```
+
```

Comment:
Isn't the default password for postgres, well, postgres? :laughing:

Reply:
Honestly I'm not sure.  I use Postgres.app and my is setup to be blank.